util/progress/progressui/printer.go

```diff
@@ -125,6 +125,11 @@ func (p *textMux) printVtx(t *trace, dgst digest.Digest) {
 	}
 	v.statusUpdates = map[string]struct{}{}
 
+	for _, w := range v.warnings[v.warningIdx:] {
+		fmt.Fprintf(p.w, "#%d WARN: %s\n", v.index, w.Short)
```

Comment:
Do you have an output example in progress ui (asciinema)? Also `detail` and `url` are not displayed in progress atm right? Are you thinking of displaying them only in debug mode?

Reply:
Yes, this just prints the (short) warning message under the vertex like a regular log from a process. The detail/url etc would need to be handled outside of the progress printing. The `DisplaySolveStatus` https://github.com/moby/buildkit/pull/2498/files#diff-c500feee78b97c9eba6731bee138b3d93ee24cf99f89733d1326b4f4d23e05cfR24 returns the list of all warnings after progress completes so whatever invoked it can do additional processing.